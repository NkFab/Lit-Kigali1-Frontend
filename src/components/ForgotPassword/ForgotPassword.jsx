import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { isEmail } from 'validator';
import {
  handleForgotPasswordForm,
  submitForgotPassword,
} from '../../redux/actions/forgotPasswordActions';
import Input from '../common/Input/Input';
import Button from '../common/Button/Button';

export class ForgotPassword extends Component {
  constructor() {
    super();
    this.state = {
      validEmail: true,
      emailError: 'Email is not valid',
    };
    this.handleInputChange = this.handleInputChange.bind(this);
    this.onSubmitButton = this.onSubmitButton.bind(this);
    this.renderErrors = this.renderErrors.bind(this);
  }

  onSubmitButton() {
    const { onSubmit, email, history } = this.props;
    this.setState({ validEmail: true });
    if (!isEmail(email)) {
      this.setState({ validEmail: false });
      return;
    }
    onSubmit({ email }).then((res) => {
      if (res.status === 201) {
        history.push('/forgot-password-message');
      }
    });
  }

  handleInputChange(e) {
    this.setState({ validEmail: true });
    const { onInputChange } = this.props;
    onInputChange({ field: e.target.name, value: e.target.value });
  }

  renderErrors() {
    let { errors } = this.props;
    const { message } = this.props;
    errors = [message, ...errors];
    return errors.map(err => <span key={err}>{err.message || err}</span>);
  }

  render() {
    const { email, submitting } = this.props;
    const { emailError, validEmail } = this.state;
    return (
      <section className="main-content middle-content">
        <form className="bg-primary-light centered-form margin-content">
          <div className="color-primary to-center">
            <h2>RESET YOUR PASSWORD</h2>
          </div>
          <div className="content-padding-h">
            <div className="form-errors">{this.renderErrors()}</div>
            <div className="content-margin-bottom">
              <div className="input primary">
                <i className="fa fa-at" />
                <Input
                  name="email"
                  type="text"
                  value={email}
                  placeholder="Enter Your Email"
                  onChange={this.handleInputChange}
                />
              </div>
              {validEmail ? '' : <div className="form-error">{emailError}</div>}
            </div>

            <Button
              classes={`primary color-white content-margin width-100 ${
                submitting ? 'loading' : ''
              }`}
              type="button"
              onClick={this.onSubmitButton}
            >
              Reset Password
            </Button>
          </div>
        </form>
      </section>
    );
  }
}

export const mapStateToProps = ({
  forgotPassword: {
    email, message, errors, submitting,
  },
}) => ({
  email,
  message,
  errors,
  submitting,
});

export const mapDispatchToProps = dispatch => ({
  onInputChange: ({ field, value }) => dispatch(handleForgotPasswordForm({ field, value })),
  onSubmit: ({ email }) => dispatch(submitForgotPassword({ email })),
});

ForgotPassword.propTypes = {
  email: PropTypes.string,
  message: PropTypes.string,
  onInputChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  errors: PropTypes.array,
  submitting: PropTypes.bool,
  history: PropTypes.any.isRequired,
};

ForgotPassword.defaultProps = {
  email: '',
  message: '',
  errors: [],
  submitting: false,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ForgotPassword);
