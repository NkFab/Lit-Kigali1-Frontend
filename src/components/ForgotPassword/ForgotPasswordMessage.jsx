import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

const ForgotPasswordMessage = ({ message }) => (
  <section className="main-content content-margin">
    <div className="container align-center">
      <p className="color-primary title-2 align-center">{message}</p>
    </div>
  </section>
);

const mapStateToProps = state => ({
  message: state.forgotPassword.successMessage,
});

ForgotPasswordMessage.propTypes = {
  message: PropTypes.string,
};
ForgotPasswordMessage.defaultProps = {
  message: '',
};

export default connect(mapStateToProps)(ForgotPasswordMessage);
