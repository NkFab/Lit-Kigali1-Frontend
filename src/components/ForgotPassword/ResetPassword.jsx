import React, { Component } from 'react';
import Button from '../common/Button/Button';
import Input from '../common/Input/Input';

class ResetPassword extends Component {
  constructor() {
    super();
    this.state = {
      newPassword: '',
      confirmNewpassword: '',
    };
    this.handleInput = this.handleInput.bind(this);
  }

  handleInput(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  render() {
    const { newPassword, confirmNewpassword } = this.state;
    return (
      <section className="main-content middle-content">
        <form className="bg-primary-light centered-form">
          <div className="color-primary to-center">
            <h2>CHANGE YOUR PASSWORD</h2>
          </div>
          <div className="content-padding-h content-margin">
            <div className="input primary content-margin">
              <i className="fa fa-lock" />
              <Input
                name="newPassword"
                value={newPassword}
                onChange={this.handleInput}
                type="password"
                placeholder="New Password"
                classes="bg-primary-light"
              />
            </div>
            <div className="input primary">
              <i className="fa fa-lock" />
              <Input
                name="confirmNewpassword"
                type="password"
                value={confirmNewpassword}
                onChange={this.handleInput}
                placeholder="Confirm New Password"
                classes="bg-primary-light"
              />
            </div>
            <Button classes="primary color-white content-margin-top width-100">
              Change Password
            </Button>
          </div>
        </form>
      </section>
    );
  }
}

export default ResetPassword;
