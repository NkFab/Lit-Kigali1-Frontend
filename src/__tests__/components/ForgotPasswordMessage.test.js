import React from 'react';
import { shallow } from 'enzyme';
import { ForgotPasswordMessage, mapStateToProps } from '../../ForgotPassword/ForgotPasswordMessage';

let wrapper;

const props = {
  message: 'Message',
};
describe('<ForgotPasswordMessage />', () => {
  test('should render the <ForgotPasswordMessage />', () => {
    wrapper = shallow(<ForgotPasswordMessage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('reducers', () => {
    test('should return `mapStateToProps`', () => {
      const initialState = {
        forgotPassword: {
          successMessage: 'message',
        },
      };
      const state = mapStateToProps(initialState);
      expect(state).toEqual({ message: 'message' });
    });
  });
});
