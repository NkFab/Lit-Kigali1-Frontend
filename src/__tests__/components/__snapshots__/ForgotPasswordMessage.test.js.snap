// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPasswordMessage /> should render the <ForgotPasswordMessage /> 1`] = `ShallowWrapper {}`;
